src/components/main/compose/messages/mod.rs

```diff
@@ -352,19 +356,28 @@ pub fn Messages(cx: Scope<Props>) -> Element {
                 }
             },
             messages.iter()
-                .zip(next_sender)
-                .zip(prev_sender)
-                .map(|((message, next_sender), prev_sender)| {
+                .enumerate()
+                .map(|(idx, message)| {
                     let message_id = message.id();
                     let conversation_id = message.conversation_id();
                     let msg_sender = message.sender();
                     let is_remote = ident.did_key() != msg_sender;
-                    let is_last = next_sender.map(|next_sender| *next_sender != msg_sender).unwrap_or(true);
-                    let is_first = prev_sender.map(|prev_sender| *prev_sender != msg_sender).unwrap_or(true);
                     let mut rg = rg.clone();
+                    let sender_picture = profile_pictures.get(&msg_sender).and_then(|pbp| pbp.clone()).unwrap_or_default();
 
+                    let is_first = if idx == 0 {
+                        false
+                    } else {
+                        let prev_message = &messages[idx - 1];
```

Comment:
I would usually suggest being careful with indexes as if the element doesnt exist or if it goes out of bounds, it would result in a panic. Looking at the code this should be fine to use but just as a caution to becareful. Whenever possible it may be best to use `messages.get` (or similar function when it comes to a vector, set or some array).